lang/parser.y

```diff
@@ -685,6 +693,91 @@ resource_field:
 		}
 	}
 ;
+edge:
+	// TODO: we could technically prevent single edge_half pieces from being
+	// parsed, but it's probably more work than is necessary...
+	// test["n1"] -> test["n2"] -> test["n3"] # chain or pair
+	edge_half_list
+	{
+		posLast(yylex, yyDollar) // our pos
+		$$.stmt = &StmtEdge{
+			EdgeHalfList: $1.edgeHalfList,
+			//Notify: false, // unused here
+		}
+	}
+//	// test["n1"].foo_send -> test["n2"].blah_recv # send/recv
+//|	edge_half_sendrecv ARROW edge_half_sendrecv
+//	{
+//		posLast(yylex, yyDollar) // our pos
+//		$$.stmt = &StmtEdge{
+//			EdgeHalfList: []*StmtEdgeHalf{
+//				$1.edgeHalf,
+//				$3.edgeHalf,
+//			},
+//			//Notify: false, // unused here, it is implied (i think)
+//		}
+//	}
+//	// test["n1"].foo_send -> test["n2"].blah_recv # send/recv
+//|	IDENTIFIER OPEN_BRACK expr CLOSE_BRACK DOT IDENTIFIER ARROW IDENTIFIER OPEN_BRACK expr CLOSE_BRACK DOT IDENTIFIER
+//	{
+//		posLast(yylex, yyDollar) // our pos
+//		e1 := &StmtEdgeHalf{
+//			Kind: $1.str,
+//			Name: $3.expr,
+//			SendRecv: $6.str,
+//		}
+//		e2 := &StmtEdgeHalf{
+//			Kind: $8.str,
+//			Name: $10.expr,
+//			SendRecv: $13.str,
+//		}
+//
+//		$$.stmt = &StmtEdge{
+//			EdgeHalfList: []*StmtEdgeHalf{
+//				e1,
+//				e2,
+//			},
+//			//Notify: false, // unused here, it is implied (i think)
+//		}
+//	}
+//;
+;
+edge_half_list:
+	/* end of list */
```

Comment:
This is the problem. You cannot do recursion this way. You are saying "a list is either 2) a list with one element appended or 1) NOTHING AT ALL". So the recursion has no anchor.

Changing this line to `edge_half` fixes the errors.

In another place you remark in a comment that it's not worth the hassle to catch "single element edge chains" at the parser level. I agree. We should just let that pass here. The fix should be valid.